### ISSUE TITLE
Clarify jacrev_and_value / jacfwd_and_value can be implemented with has_aux in documentation

### ISSUE THREAD

**Issue by AlphaBetaGamma96** [pytorch/functorch#502]
Hello! 

I was reading issue #342 and I noticed that you have mentioned if you were going to have a `jacrev_and_value` and `jacfwd_and_value`, and I thought I'd ask for it as it's quite useful when computing 2nd order gradients within the log-domain. 

To just explain further let's say you're computing `torch.linalg.det` and want to find the Laplacian (trace of Hessian) of it with respect to its inputs, but you compute the output within the log-domain via `torch.linalg.slogdet` for numerical stability. Then the Laplacian of `torch.linalg.det` would be equal to the laplacian of `torch.linalg.slogdet` plus the square of the jacobian of `torch.linalg.slogdet`. I've attached an image of this mathematically just to make what I've said above clear. As this generalises to any function computed within the log-domain!

![image](https://user-images.githubusercontent.com/69877955/154353169-53b934ee-1533-4857-8f09-26cc8c8f75f0.png)

This type of function is needed when creating models of many-body problems in Quantum Mechanics, so having a function that returns both the jacobian and its value would be highly useful! 

Thank you!

**Comment by zou3519** [comment]
@AlphaBetaGamma96 Does using jacrev and jacfwd with the `has_aux = True` cover this case? For example, to compute the jacobian of f and get f(x):

```
def g(x):
  result = f(x)
  return result, result

jacobian, result = jacrev(g, has_aux=True)(x)
```

**Comment by AlphaBetaGamma96** [comment]
Hi @zou3519 it does indeed! Apologises as I must've missed the `has_aux` flag! 

I assume `has_aux` hasn't been implemented for `jacfwd` so far? 
```
import torch 
from functorch import jacrev, jacfwd, vmap

def func(x):
    result = torch.sin(x)
    return result, result

x = torch.randn(10)

df_dx, fx = vmap(jacrev(func, has_aux=True), in_dims=(0))(x)  #works
#df_dx, fx = vmap(jacfwd(func, has_aux=True), in_dims=(0))(x) #misses `has_aux` arg

print("1st derivative: ",torch.allclose(df_dx, x.cos())) #returns True
print("output: ", torch.allclose(fx, x.sin())) #returns True
```

Also, I assume the reason why the `result` is duplicated is that one copy is differentiated whereas the other copy is just ignored and returned as the value? Is that a correct understanding of copying the output and the `has_aux` flag?

Thank you!

**Comment by vfdev-5** [comment]
> I assume has_aux hasn't been implemented for jacfwd so far?

@AlphaBetaGamma96 it is very recent addition: https://github.com/pytorch/functorch/blob/588410bd2950954eddf2571e88b4bbee7d4332f5/functorch/_src/eager_transforms.py#L791

> Also, I assume the reason why the result is duplicated is that one copy is differentiated whereas the other copy is just ignored and returned as the value? Is that a correct understanding of copying the output and the has_aux flag?

Yes, if has_aux is True, function's output is expected as (output, aux) and the first part is used for jacobian computations and the second is simply returned as it is. 
As func return `return result, result` without explicit copying of `result` for aux, I hope there is no side-effect (probably, no) between result as output and same tensor as aux.

**Comment by zou3519** [comment]
Let's leave this open to add documentation to jacrev/jacfwd about how to compute the output of the function at the same time. You're not the first to have asked, @AlphaBetaGamma96.
